### PR TITLE
Add loki image and tests.

### PR DESCRIPTION
**Pull request by dlorenc**
## Chainguard Images Pull Request Template

- [X] The Image is smaller in size than its common public counterpart.

Notes:

### Image Vulnerabilities

- [x] The Grype vulnerability scan returned 0 CVE(s).
- [ ] The Grype vulnerability scan returned > 0 CVE(s) (please explain in the notes).

Notes:

### Basic Testing - Package/Application

- [X] The application is accessible to the user/cluster/etc. after start-up.

Notes:

### Helm

- [X] A Helm chart was not provided.

Notes:

### Processor Architectures

- [X] The image was built and tested for x86_64.
- [X] The image was built and tested for aarch64.

Notes:

### Functional Testing + Documentation

- [X] Functional tests have been included and the tests are passing.  All tests have been documnted in the notes section.

Notes:

### Environment Testing + Documentation

- [X] There has not been a request and/or there is no indication that this image needs tested on a public cloud provider.

Notes:

### Version

- [X] The package version is the latest version of the package.  The latest tag points to this version.

Notes:

### Dev Tag Availability

- [X] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')

Notes:

### Access Control + Authentication

- [X] The image runs as `loki` and GID/UID are set to 65532 or upstream default

### ENTRYPOINT

- [X] applications/servers/utilities set to call main program with no arguments e.g. [redis-server]

### CMD

- [X] For server applications give arguments to start in daemon mode (may be empty)

### Environment Variables

- [X] Environment variables not added and not required.

### SIGTERM

- [X] The image responds to SIGTERM (e.g., `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`)

### Logs

- [X] Error logs write to stderr and normal logs to stdout. Logs DO NOT write to file.

### Documentation - README

- [X] A README file has been provided and it follows the README template.

### CODE REVIEW

File: images/loki/configs/latest.apko.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (mattmoor) [inline images/loki/configs/latest.apko.yaml:1]: nit: extra space

File: images/loki/tests/01-version.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o errtrace -o pipefail -x
+
+docker run --rm $IMAGE_NAME --version
```
Comment (mattmoor) [inline images/loki/tests/01-version.sh:5]: Can we just inline this into `main.tf`?  Should be a one-liner

File: images/loki/tests/02-use.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+set -o errexit -o nounset -o errtrace -o pipefail -x
+
+echo "${IMAGE_NAME}"
+
+
+id=$(docker run -d --rm "${IMAGE_NAME}")
+sleep 10
+trap "docker kill ${id}" EXIT
+
+docker logs "${id}" 2>&1 | grep "Loki started"
```
Comment (mattmoor) [inline images/loki/tests/02-use.sh:12]: Is there a Helm chart for this, or a way we can actually test the log streaming?

**Review by mattmoor — COMMENTED** [review]
I don't see this on endoflife.date, do we know what the upstream support policy for Loki is?

File: images/loki/tests/02-use.sh (hunk 3 above)
Comment (mattmoor) [reply images/loki/tests/02-use.sh:12]: Look like: https://github.com/grafana/loki/tree/main/production/helm

Comment (mattmoor) [reply images/loki/tests/02-use.sh:12]: Actually that may be deprecated in favor of: https://github.com/grafana/helm-charts/tree/main/charts/loki-stack?

**Review by imjasonh — APPROVED** [review]
Season 2 is coming out in October!

File: images/loki/tests/02-use.sh (hunk 3 above)
Comment (dlorenc) [reply images/loki/tests/02-use.sh]: I think they just moved the charts, trying to get that one running.

File: images/loki/tests/main.tf [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+  })]
+}
+
+data "oci_exec_test" "version" {
+  digest = var.digest
+  script = "docker run --rm $IMAGE_NAME --version"
+}
+
+data "oci_exec_test" "compile" {
```
Comment (mattmoor) [inline images/loki/tests/main.tf:55]: nit:
```suggestion
data "oci_exec_test" "use" {
```

File: images/loki/tests/main.tf [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+data "oci_exec_test" "test" {
+  for_each = toset(fileset(path.module, "*.sh"))
+
+  digest = var.digest
+  script = "${path.module}/${each.value}"
+}
```
Comment (mattmoor) [inline images/loki/tests/main.tf:19]: I think this is redundant now with the resource below?

File: images/loki/tests/main.tf [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+resource "helm_release" "loki" {
```
Comment (mattmoor) [inline images/loki/tests/main.tf:21]: Is there any good way we can check this once it is up?  Sort of like we use the metrics server here with `kubectl top`: https://github.com/chainguard-images/images/blob/fc964c7de0ebdaa17d242efb517b3052d11b0c78/images/metrics-server/tests/main.tf#L41

Comment (mattmoor) [reply images/loki/tests/main.tf:21]: I've learned that helm charts vary a lot in terms of what "successfully applied" means.  Some return after `kubectl apply` (iirc metrics server) and others do a proper readiness check (iirc prometheus-stack).  If we think this does reasonable checking, then we can maybe forgo additional checks, but we should at least confirm a reasonable readiness check.

**Review by mattmoor — APPROVED** [review]

File: images/loki/tests/main.tf (hunk 6 above)
Comment (dlorenc) [reply images/loki/tests/main.tf:21]: > I've learned that helm charts vary a lot in terms of what "successfully applied" means. Some return after `kubectl apply` (iirc metrics server) and others do a proper readiness check (iirc prometheus-stack). If we think this does reasonable checking, then we can maybe forgo additional checks, but we should at least confirm a reasonable readiness check.

I checked, this one waits for readiness!

**Review by mattmoor — APPROVED** [review]

**Pull request merged**